java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/processfunction/SlidingWindowKeyedProcessFunction.java

```diff
@@ -188,9 +232,10 @@ public void onTimer(
                 if (timestamp - descriptor.windowSizeMs < rowTime) {
                     break;
                 }
-                Row curRow = state.timestampToRow.get(rowTime);
-                descriptor.aggFunc.retractAccumulator(
-                        accumulatorState, curRow.getField(descriptor.fieldName));
+                for (Row curRow : state.timestampToRows.get(rowTime)) {
```

Comment:
It would be better to rename `curRow` to be consistent with `rowToAdd` above.

Maybe rename both variables as `row`.